File: php/lib/steam/packets/M2C_ISVALIDMD5_Packet.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2010-2011, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Packets
 */

require_once STEAM_CONDENSER_PATH . 'steam/packets/SteamPacket.php';

/**
 * A packet of type M2S_ISVALIDMD5 is used by the master server to provide a
 * challenge number to a game server
 *
 * @package Steam Condenser (PHP)
 * @subpackage Packets
 */
class M2C_ISVALIDMD5_Packet extends SteamPacket
{

    /**
     * @var int
     */
    private $challenge;

    /**
     * Creates a new response packet with the data from the master server
     *
     * @param string $data The packet data sent by the master server
     */
    public function __construct($data) {
        parent::__construct(SteamPacket::M2C_ISVALIDMD5_HEADER, $data);

        $this->contentData->getByte();
        $this->challenge = $this->contentData->getUnsignedLong();
    }

    /**
     * Returns the challenge number to use for master server communication
     *
     * @return long The challenge number
     */
    public function getChallenge() {
        return $this->challenge;
    }
}
?>

File: ruby/lib/steam/packets/m2c_isvalidmd5_packet.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2011, Sebastian Staudt

require 'steam/packets/steam_packet'

# This packet class represents a M2S_ISVALIDMD5 response replied by a master
# server
#
# It is used to provide a challenge number to a game server
#
# @author Sebastian Staudt
# @see MasterServer#challenge
class M2C_ISVALIDMD5_Packet

  include SteamPacket

  # Returns the challenge number to use for master server communication
  #
  # @return [Fixnum] The challenge number
  attr_reader :challenge

  # Creates a new M2S_ISVALIDMD5 response object based on the given data
  #
  # @param [String] data The raw packet data replied from the server
  def initialize(data)
    super M2C_ISVALIDMD5_HEADER, data

    @content_data.byte
    @challenge = @content_data.long
  end

end
